scripts/run_experiment.sh

```diff
@@ -38,29 +43,38 @@ cat << EOF > $GET_RESULTS_SH
 #SBATCH --time=00:00:59
 #SBATCH -A plgccbmc11-cpu
 
-SACCT_RESULT="\$(sacct -j $COLLECT_ID --format State,End)"
-echo \$SACCT_RESULT
-arrIN=(\${SACCT_RESULT//;/ })
-STATE=\$(echo \${arrIN[4]})
-if [[ "\$STATE" == "COMPLETED" ]]
-then
-    END=\$(echo \${arrIN[5]})
-    T_END=\$(date --date="\$END" +"%s%N")
-elif [[ "\$STATE" == "FAILED" ]]
-then
-    echo "Experiment failed, exiting..."
-    exit 1
-fi
+while true
+do
+    SACCT_RESULT="\$(sacct -j $COLLECT_ID --format State,End)"
+    echo \$SACCT_RESULT
+    arrIN=(\${SACCT_RESULT//;/ })
+    STATE=\$(echo \${arrIN[4]})
+    if [[ "\$STATE" == "COMPLETED" ]]
+    then
+        END=\$(echo \${arrIN[5]})
+        T_END=\$(date --date="\$END" +"%s%N")
 
-T_EXEC_SECS=\$(( (T_END - $T_START) / 1000000000 ))
-T_EXEC_NANS=\$(( (T_END - $T_START) % 1000000000 ))
+        echo $T_START
+        echo \$T_END
+        T_EXEC_SECS=\$(( (T_END - $T_START) / 1000000000 ))
+        T_EXEC_NANS=\$(( (T_END - $T_START) % 1000000000 ))
 
-echo "Run with ${N} nodes took: \${T_EXEC_SECS}s \${T_EXEC_NANS}ns"
-echo ""
-printf "%d.%09d" \${T_EXEC_SECS} \${T_EXEC_NANS} > ${RUN_DIR}/time.txt
-echo "${N},\$(cat ${RUN_DIR}/time.txt)" >> ${BASE_DIR}/output/raw/times.csv
+        echo "Run with ${N} nodes took: \${T_EXEC_SECS}s \${T_EXEC_NANS}ns"
+        echo ""
+        printf "%d.%09d" \${T_EXEC_SECS} \${T_EXEC_NANS} > ${RUN_DIR}/time.txt
+        echo "${N},\$(cat ${RUN_DIR}/time.txt)" >> ${BASE_DIR}/output/raw/times.csv
 
-rm -rf $run_path
+        rm -rf $run_path
+        exit 0
+    elif [[ "\$STATE" == "FAILED" ]]
+    then
+        echo "Experiment failed, exiting..."
+        exit 1
+    else
+        echo "Waiting for task to finish - I was started too quickly"
+    fi
+    sleep 0.1
```

Comment:
`sleep` akceptuje ułamki sekund? Za bardzo przenośne to to chyba nie jest, zresztą nawalanie co 0.1 sekundy logami też chyba nie jest potrzebne

```suggestion
    sleep 1
```

Reply:
W naszych eksperymentach 1 sekunda to nawet ponad 10% czasu wykonania, a `0.1` jak na razie wystarczało na `odlagowanie` systemu, ale trzeba zbadać sprawę w przyszłości

Reply:
Nie commituje tej sugestii

Reply:
Przekonujące. Ja wohl mein Obersturmgruppenfuhrer